tests/cbmc/proofs/s2n_stuffer_read_bytes/s2n_stuffer_read_bytes_harness.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "api/s2n.h"
+#include "stuffer/s2n_stuffer.h"
+#include <assert.h>
+#include <cbmc_proof/proof_allocators.h>
+#include <cbmc_proof/cbmc_utils.h>
+#include <cbmc_proof/make_common_datastructures.h>
+
+void s2n_calculate_stacktrace() {}
+
+void s2n_stuffer_read_bytes_harness() {
+    struct s2n_stuffer *stuffer = cbmc_allocate_s2n_stuffer();
+    __CPROVER_assume(s2n_stuffer_is_valid(stuffer));
+    struct s2n_stuffer old_stuffer = *stuffer;
+    struct store_byte_from_buffer old_byte;
+
+    save_byte_from_blob(&stuffer->blob, &old_byte);
+
+    uint32_t length;
+    uint8_t *dst = can_fail_malloc(length);
+   
+    if (s2n_stuffer_read_bytes(stuffer, dst, length) == 0) {
+        assert(stuffer->read_cursor == old_stuffer.read_cursor + length);
+        assert_byte_from_blob_matches(&stuffer->blob, &old_byte); 
```

Comment:
This should be true whether the read happened or not. Move it below.

Reply:
I see. I was comparing the wrong values. I've updated the proof to compare that the byte was read into the destination correctly.